discojs/src/training/trainer.ts

```diff
@@ -94,24 +111,25 @@ export class Trainer<D extends DataType, N extends Network> {
     void
   > {
     const totalRound = Math.trunc(this.#epochs / this.#roundDuration);
-    let previousRoundWeights: WeightsContainer | undefined;
     for (let round = 0; round < totalRound; round++) {
+
       await this.#client.onRoundBeginCommunication();
 
+      // in onRoundEndCommunication function, allow the case where there's no input
+      // since even in the case when the client doesn't run the training it has to still get the updated weights
       yield this.#runRound(dataset, validationDataset);
 
-      let localWeights = this.model.weights;
-      if (this.#privacy !== undefined)
-        localWeights = await applyPrivacy(
-          previousRoundWeights,
-          localWeights,
-          this.#privacy,
-        );
+      const localWeights = this.model.weights;
 
+      // Get the updated weights
       const networkWeights =
-        await this.#client.onRoundEndCommunication(localWeights);
+      await this.#client.onRoundEndCommunication(localWeights);
```

Comment:
confusing to have it at the same identation, it looks like a statement rather than an assignement
```suggestion
        await this.#client.onRoundEndCommunication(localWeights);
```